torchgeo/trainers/detection.py

```diff
@@ -7,24 +7,17 @@
 
 import pytorch_lightning as pl
 import torch
-import torchmetrics
 import torchvision
 from packaging.version import parse
 from torch import Tensor
 from torch.optim.lr_scheduler import ReduceLROnPlateau
 from torch.utils.data import DataLoader
+from torchmetrics.detection.mean_ap import MeanAveragePrecision
 from torchvision.models.detection import FasterRCNN
 from torchvision.models.detection.backbone_utils import resnet_fpn_backbone
 from torchvision.models.detection.rpn import AnchorGenerator
 from torchvision.ops import MultiScaleRoIAlign
 
-if parse(torchmetrics.__version__) >= parse("0.8.0"):
-    from torchmetrics.detection.mean_ap import MeanAveragePrecision
-else:
-    from torchmetrics.detection.map import (  # type: ignore[no-redef]
-        MeanAveragePrecision,
-    )
```

Comment:
In hindsight, it really wouldn't be that hard to support older torchmetrics. We would basically copy this same structure to the ~3 places where we import classification metrics. But I also don't see a huge advantage of retaining support for older torchmetrics.

Reply:
Also, if we decide to do this for torchvision too, we can drop our packaging dependency.

Reply:
Okay, yeah, it would be too much work to support both.